sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java

```diff
@@ -355,6 +356,44 @@ static TwoTableJoin buildTwoTableJoinTree(MultiSourceSelect mss) {
         return join;
     }
 
+
+    /**
+     * Build outputs of joins not in the outer level of the tree
+     * based on the where queries that must be applied later on
+     * in case the current outputs are empty (case of count(*))
+     * e.g.:
+     * select count(*) from t1, t2, t3 where t1.a = t2.b and t2.b = t3.c
+     *
+     *                   join
+     * outputs=t1[id]   /   \
+     *      |          /     \
+     *      +------> join    t3
+     *               / \
+     *              /   \
+     *             t1   t2
+     */
+    private static void buildJoinOutputs(Map<Set<QualifiedName>, Symbol> splitQuery,
```

Comment:
maybe rename to somethingl ike `updateQSOutputs` or `extendQSOutputs` to at least indicate where the side effect happens.